flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/DefaultStateTransitionManager.java

```diff
@@ -0,0 +1,434 @@
+
+    @VisibleForTesting final Duration cooldownTimeout;
+    @Nullable @VisibleForTesting final Duration resourceStabilizationTimeout;
+    @VisibleForTesting final Duration maxTriggerDelay;
+
+    DefaultStateTransitionManager(
+            Temporal initializationTime,
+            StateTransitionManager.Context transitionContext,
+            Duration cooldownTimeout,
+            @Nullable Duration resourceStabilizationTimeout,
+            Duration maxTriggerDelay) {
+        this(
+                initializationTime,
+                Instant::now,
+                transitionContext,
+                cooldownTimeout,
+                resourceStabilizationTimeout,
+                maxTriggerDelay);
+    }
+
+    @VisibleForTesting
+    DefaultStateTransitionManager(
+            Temporal initializationTime,
+            Supplier<Temporal> clock,
+            StateTransitionManager.Context transitionContext,
+            Duration cooldownTimeout,
+            @Nullable Duration resourceStabilizationTimeout,
+            Duration maxTriggerDelay) {
+
+        this.clock = clock;
+        this.maxTriggerDelay = maxTriggerDelay;
+        this.cooldownTimeout = cooldownTimeout;
+        this.resourceStabilizationTimeout = resourceStabilizationTimeout;
+        this.transitionContext = transitionContext;
+        this.scheduledFutures = new ArrayList<>();
+        this.phase = new Cooldown(initializationTime, clock, this, cooldownTimeout);
+    }
+
+    @Override
+    public void onChange() {
+        phase.onChange();
+    }
+
+    @Override
+    public void onTrigger() {
+        phase.onTrigger();
+    }
+
+    @Override
+    public void close() {
+        scheduledFutures.forEach(future -> future.cancel(true));
+        scheduledFutures.clear();
+    }
+
+    @VisibleForTesting
+    Phase getPhase() {
+        return phase;
+    }
+
+    private void progressToIdling() {
+        progressToPhase(new Idling(clock, this));
+    }
+
+    private void progressToStabilizing(Temporal firstChangeEventTimestamp) {
+        progressToPhase(
+                new Stabilizing(
+                        clock,
+                        this,
+                        resourceStabilizationTimeout,
+                        firstChangeEventTimestamp,
+                        maxTriggerDelay));
+    }
+
+    private void progressToStabilized(Temporal firstChangeEventTimestamp) {
+        progressToPhase(new Stabilized(clock, this, firstChangeEventTimestamp, maxTriggerDelay));
+    }
+
+    private void triggerTransitionToSubsequentState() {
+        progressToPhase(new Transitioning(clock, this));
+        transitionContext.transitionToSubsequentState();
+    }
+
+    private void progressToPhase(Phase newPhase) {
+        Preconditions.checkState(
+                !(phase instanceof Transitioning),
+                "The state transition operation has already been triggered.");
+        LOG.debug("Transitioning from {} to {}.", phase, newPhase);
```

Comment:
This is logged on DEBUG (before it was INFO IIRC)
```
Transitioning from org.apache.flink.runtime.scheduler.adaptive.DefaultStateTransitionManager$Stabilizing@79930ca7 to org.apache.flink.runtime.scheduler.adaptive.DefaultStateTransitionManager$Stabilized@4db16ef7.
```
without job ID (neither in log message nor in MDC).
I think this is a regression

Reply:
Thanks!